src/main/java/io/swagger/service/IMDBService.java

```diff
@@ -27,7 +27,7 @@ public TitleData getMediaById(String id) {
 
         ResponseEntity<TitleData> response = restTemplate.getForEntity(url, TitleData.class);
 
-        if (response.getStatusCode() == HttpStatus.OK) {
+        if (response.getStatusCode() == HttpStatus.OK && response.getBody() != null && response.getBody().getId() != null) {
             return response.getBody();
```

Comment:
Is there not an issue with calling getBody twice? Some HTTP clients use a stream buffer that will error out if you call getBody twice (I guess I'll see for sure when I test it out).